matchmake-extension/auto_matchmake_with_search_criteria_postpone.go

```diff
@@ -0,0 +1,126 @@
+package matchmake_extension
+
+import (
+	nex "github.com/PretendoNetwork/nex-go"
+	common_globals "github.com/PretendoNetwork/nex-protocols-common-go/globals"
+	match_making_types "github.com/PretendoNetwork/nex-protocols-go/match-making/types"
+	matchmake_extension "github.com/PretendoNetwork/nex-protocols-go/matchmake-extension"
+	"github.com/PretendoNetwork/nex-protocols-go/notifications"
+	notifications_types "github.com/PretendoNetwork/nex-protocols-go/notifications/types"
+)
+
+func autoMatchmakeWithSearchCriteria_Postpone(err error, client *nex.Client, callID uint32, lstSearchCriteria []*match_making_types.MatchmakeSessionSearchCriteria, anyGathering *nex.DataHolder, message string) {
+	server := commonMatchmakeExtensionProtocol.server
+
+	// A client may disconnect from a session without leaving reliably,
+	// so let's make sure the client is removed from the session
+	common_globals.RemoveConnectionIDFromAllSessions(client.ConnectionID())
+
+	var matchmakeSession *match_making_types.MatchmakeSession
+	anyGatheringDataType := anyGathering.TypeName()
+
+	if anyGatheringDataType == "MatchmakeSession" {
+		matchmakeSession = anyGathering.ObjectData().(*match_making_types.MatchmakeSession)
+	}
+
```

Comment:
Should we also cleanup the search criteria parameters here just in case?

Reply:
I was going to mention this in my PR as well but instead of duplicating the comment I'll just add this here:

If this type is ever not `MatchmakeSession` then the `matchmakeSession` variable will remain in its zero state and will have nil pointers. I assume that because you're checking the type name that means there's a chance it could be a different type, those cases should probably be handled (even if it's an `else` statement here for now that logs a warning and bails or something)

Reply:
Yeah, probably a wise idea.